Source/Core/Core/PowerPC/JitArm64/JitArm64_RegCache.cpp

```diff
@@ -298,18 +298,52 @@ ARM64Reg Arm64FPRCache::R(u32 preg, bool only_lower)
 			// Change it over to a full 128bit register
 			reg.LoadToReg(reg.GetReg());
 		}
+		else if (type == REG_DUP)
+		{
+			// We already only have the lower 64bits
+			// Don't do anything
+		}
 		return reg.GetReg();
 	}
 	break;
+	case REG_DUP:
+	{
+		ARM64Reg host_reg = reg.GetReg();
+		if (type == REG_REG)
+		{
+			// We are requesting a full 128bit register
+			// but we are only available in the lower 64bits
+			// Duplicate to the top and change over
+			m_float_emit->INS(64, host_reg, 1, host_reg, 0);
+			reg.LoadToReg(host_reg);
+		}
+		else if (type == REG_LOWER_PAIR)
+		{
+			// We are only requesting the lower 64bits of a pair
+			// We've got to be careful in this instance
+			// Store our current duplicated high bits to the file
+			// then convert over to a lower reg
+			if (reg.IsDirty())
+				m_float_emit->STR(64, INDEX_UNSIGNED, host_reg, X29, PPCSTATE_OFF(ps[preg][1]));
```

Comment:
There's actually a very particular instance that can cause issues.
Without this store, if we have a REG_DUP that changes over to a REG_LOWER_PAIR and then changes over to a REG_REG then it will load garbage data in to the high 64bits.